// playwright-test-framework-advanced/src/utils/Logger.js
import winston from 'winston';

export class Logger {
  static getInstance() {
    if (!Logger.instance) {
      Logger.instance = winston.createLogger({
        level: process.env.LOG_LEVEL || 'info',
        format: winston.format.combine(
          winston.format.timestamp(),
          winston.format.errors({ stack: true }),
          winston.format.json()
        ),
        transports: [
          new winston.transports.Console({
            format: winston.format.combine(
              winston.format.colorize(),
              winston.format.simple()
            )
          }),
          new winston.transports.File({ 
            filename: 'logs/error.log', 
            level: 'error' 
          }),
          new winston.transports.File({ 
            filename: 'logs/combined.log' 
          })
        ]
      });
    }
    return Logger.instance;
  }

  static info(message, meta) {
    Logger.getInstance().info(message, meta);
  }

  static error(message, meta) {
    Logger.getInstance().error(message, meta);
  }

  static warn(message, meta) {
    Logger.getInstance().warn(message, meta);
  }

  static debug(message, meta) {
    Logger.getInstance().debug(message, meta);
  }
}
